google/cloud/forseti/enforcer/gce_firewall_enforcer.py

```diff
@@ -375,7 +361,19 @@ def add_rule(self, rule, network_name=None):
         if 'disabled' not in new_rule:
             new_rule['disabled'] = self.DEFAULT_DISABLED
 
-        if self._check_rule_before_adding(new_rule):
+        error = rv.validate_gcp_rule(new_rule)
+        if error:
+            raise InvalidFirewallRuleError(error)
+
+        if rule['name'] in self.rules:
+            raise DuplicateFirewallRuleNameError(
+                'Rule %s already defined in rules: %s' % (
+                    rule['name'], ', '.join(sorted(self.rules.keys()))))
+
+        if self._add_rule_callback:
```

Comment:
(nit) Possible improvement in logic here.

```
add_rule = self._add_rule_callback(new_rule) if self._add_rule_callback else True
if add_rule:
	self.rules[new_rule['name']] = new_rule
```

Reply:
Agreed